src/core/ready-no-deferred.js

```diff
@@ -0,0 +1,109 @@
+define( [
+	"../core",
+	"../var/document"
+], function( jQuery, document ) {
+
+var readyCallbacks = [],
+	readyFiring = false,
+	whenReady = function( fn ) {
+		readyCallbacks.push( fn );
+	},
+	executeReady = function( fn ) {
+
+		// Prevent errors from freezing future callback execution (gh-1823)
+		// Not backwards-compatible as this does not execute sync
+		window.setTimeout( function() {
+			fn.call( document, jQuery );
```

Comment:
This PR is the _introduction_ of `jQuery.ready.then`... why start off on the wrong foot when we don't have to? It doesn't even make sense to create a ticket referencing behavior that hasn't landed. But I guess I'll put some text here just in case:
> Per [Promises/A+](https://promisesaplus.com/#point-35) and [ES2015](http://www.ecma-international.org/ecma-262/6.0/#sec-promisereactionjob), `then` callbacks should be called without a `this` context. And although `jQuery.Deferred#*With` methods provide context to callbacks, we are moving in the direction of increased standards compliance, and should stop providing `document` context to `jQuery.ready` callbacks while we're already making backwards-incompatible changes to `jQuery.ready`.